src/main/kotlin/com/cjbooms/fabrikt/generators/GeneratorUtils.kt

```diff
@@ -168,10 +170,12 @@ object GeneratorUtils {
 
         val parameters = mergeParameters(pathParameters, parameters)
             .map {
+                // For backwards compatibility, only non-ref parameters get the name suffix.
+                val nameSuffix = if (Overlay.of(it).pathFromRoot.startsWith("/paths")) it.name.pascalCase() else ""
                 RequestParameter(
                     it.name,
                     it.description,
-                    toModelType(basePackage, KotlinTypeInfo.from(it.schema), isNullable(it)),
+                    toModelType(basePackage, KotlinTypeInfo.from(it.schema, nameSuffix = nameSuffix), isNullable(it)),
```

Comment:
**question (non-blocking):** Why is it required to write it out `nameSuffix = nameSuffix ` instead of just a single `nameSuffix` ?

Reply:
The called function has other parameters which we do not want to specify. We could write `from(it.schema, "", "", nameSuffix)` instead for semantically equicalent code. But that'd be arguably worse from a code quality.perspective.

Kotlin, unlike e.g. Typescript in some contexts, doesn't infer parameter names from variable names.

Reply:
Thank you very much for the explanation :)